### PR TITLE
fix: use remote address instead of client ip

### PR DESCRIPTION
**Pull request by starsz**
Please answer these questions before submitting a pull request, **or your PR will get closed**.

**Why submit this pull request?**
Now, we use the gin's context.ClientIP.This function will parse the IP from the header `X-Forwarded-For` and `X-Real-Ip`
which will bypass the ipfilter middleware.

- [x] Bugfix
- [ ] New feature provided
- [ ] Improve performance
- [ ] Backport patches

**What changes will this PR take into?**

Use remote address instead of client ip.

**Checklist:**

- [x] Did you explain what problem does this PR solve? Or what new features have been added?
- [x] Have you added corresponding test cases?
- [ ] Have you modified the corresponding document?
- [ ] Is this PR backward compatible? If it is not backward compatible, please discuss on the mailing list first

### CODE REVIEW

File: api/internal/filter/ip_filter.go [hunk 1]
```diff
@@ -81,7 +82,10 @@ func checkIP(ipStr string, ips map[string]bool, subnets []*subnet) bool {
 func IPFilter() gin.HandlerFunc {
 	ips, subnets := generateIPSet(conf.AllowList)
 	return func(c *gin.Context) {
-		ipStr := c.ClientIP()
+		var ipStr string
+		if ip, _, err := net.SplitHostPort(strings.TrimSpace(c.Request.RemoteAddr)); err == nil {
```
Comment (tokers) [inline api/internal/filter/ip_filter.go:86]: Why need `strings.TrimSpace`?

Comment (starsz) [reply api/internal/filter/ip_filter.go:86]: Referer from gin:

https://github.com/gin-gonic/gin/blob/ee4de846a894e9049321e809d69f4343f62d2862/context.go#L760-L763

Comment (tokers) [reply api/internal/filter/ip_filter.go:86]: All right.

**Review by tokers — APPROVED** [review]

**Review by nic-chen — APPROVED** [review]

File: api/internal/filter/ip_filter_test.go [hunk 2]
```diff
@@ -55,4 +56,22 @@ func TestIPFilter_Handle(t *testing.T) {
 	})
 	w = performRequest(r, "GET", "/test")
 	assert.Equal(t, 200, w.Code)
+
+	// should forbidden
+	conf.AllowList = []string{"8.8.8.8"}
```
Comment (moonming) [inline api/internal/filter/ip_filter_test.go:61]: please use 127.0.0.1 for test

**Review by moonming — CHANGES_REQUESTED** [review]

File: api/internal/filter/ip_filter_test.go (hunk 2 above)
Comment (starsz) [reply api/internal/filter/ip_filter_test.go]: Ok. Updated.

**Review by moonming — APPROVED** [review]

**Pull request merged**